src/DataProtection/DataProtection/test/Microsoft.AspNetCore.DataProtection.Tests/XmlEncryption/XmlEncryptionExtensionsTests.cs

```diff
@@ -49,6 +49,99 @@ public void DecryptElement_RootNodeRequiresDecryption_Success()
         XmlAssert.Equal("<newNode />", retVal);
     }
 
+    [Fact]
+    public void DecryptElement_CustomType_TypeNameResolverNotCalled()
+    {
+        // Arrange
+        var decryptorTypeName = typeof(MyXmlDecryptor).AssemblyQualifiedName;
+
+        var original = XElement.Parse(@$"
+                <x:encryptedSecret decryptorType='{decryptorTypeName}' xmlns:x='http://schemas.asp.net/2015/03/dataProtection'>
+                  <node />
+                </x:encryptedSecret>");
+
+        var mockActivator = new Mock<IActivator>();
+        mockActivator.ReturnDecryptedElementGivenDecryptorTypeNameAndInput(decryptorTypeName, "<node />", "<newNode />");
+        var mockTypeNameResolver = mockActivator.As<ITypeNameResolver>();
+
+        var serviceCollection = new ServiceCollection();
```

Comment:
I must be missing something, but why is there a service collection?  Why not just use the activator directly?  Is something retrieving it from DI in the background?

Reply:
All the tests in this file do it. I went with the flow 😄